src/app-gocardless/services/tests/gocardless-service.spec.js

```diff
@@ -556,64 +470,56 @@ describe('goCardlessService', () => {
       );
       expect(getBalancesSpy).toBeCalledTimes(1);
     });
-
-    it('handle error if status_code present in the response', async () => {
-      getBalancesSpy.mockResolvedValue(mockUnknownError);
-
-      await expect(() =>
-        goCardlessService.getBalances(accountId),
-      ).rejects.toThrow(UnknownError);
-    });
   });
 });
 
 describe('#handleGoCardlessError', () => {
   it('throws InvalidInputDataError for status code 400', () => {
-    const response = { status_code: 400 };
+    const response = { response: { status: 400 } };
     expect(() => handleGoCardlessError(response)).toThrow(
       InvalidInputDataError,
     );
   });
 
   it('throws InvalidGoCardlessTokenError for status code 401', () => {
-    const response = { status_code: 401 };
+    const response = { response: { status: 401 } };
     expect(() => handleGoCardlessError(response)).toThrow(
       InvalidGoCardlessTokenError,
     );
   });
 
   it('throws AccessDeniedError for status code 403', () => {
-    const response = { status_code: 403 };
+    const response = { response: { status: 403 } };
     expect(() => handleGoCardlessError(response)).toThrow(AccessDeniedError);
   });
 
   it('throws NotFoundError for status code 404', () => {
-    const response = { status_code: 404 };
+    const response = { response: { status: 404 } };
     expect(() => handleGoCardlessError(response)).toThrow(NotFoundError);
   });
 
   it('throws ResourceSuspended for status code 409', () => {
-    const response = { status_code: 409 };
+    const response = { response: { status: 409 } };
     expect(() => handleGoCardlessError(response)).toThrow(ResourceSuspended);
   });
 
   it('throws RateLimitError for status code 429', () => {
-    const response = { status_code: 429 };
+    const response = { response: { status: 429 } };
     expect(() => handleGoCardlessError(response)).toThrow(RateLimitError);
   });
 
   it('throws UnknownError for status code 500', () => {
-    const response = { status_code: 500 };
+    const response = { response: { status: 500 } };
     expect(() => handleGoCardlessError(response)).toThrow(UnknownError);
   });
 
   it('throws ServiceError for status code 503', () => {
-    const response = { status_code: 503 };
+    const response = { response: { status: 503 } };
     expect(() => handleGoCardlessError(response)).toThrow(ServiceError);
   });
 
   it('does not throw an error for status code 200', () => {
-    const response = { status_code: 200 };
+    const response = { response: { status: 200 } };
     expect(() => handleGoCardlessError(response)).not.toThrow();
   });
```

Comment:
This part is now changed, because we only pass in the response when it throws, not on a successful (200) call.